recipes/full_finetune_distributed.py

```diff
@@ -15,25 +16,21 @@
 from omegaconf import DictConfig, ListConfig
 
 from torch import nn
-from torch.distributed import init_process_group
-from torch.distributed.fsdp import (
-    CPUOffload,
-    FullOptimStateDictConfig,
-    FullStateDictConfig,
-    FullyShardedDataParallel as FSDP,
-    StateDictType,
+from torch.distributed import destroy_process_group, init_process_group
+from torch.distributed._composable.fsdp import CPUOffloadPolicy, fully_shard
+from torch.distributed.algorithms._checkpoint.checkpoint_wrapper import (
```

Comment:
If we're thinking about the same thing, it may have been the suggestion to move to [torch.utils.checkpoint](https://pytorch.org/docs/stable/checkpoint.html) instead of relying on distributed APIs for AC. If it's all the same to you I would punt on that here